Modules/rotatingtree.c

```diff
@@ -10,17 +16,20 @@
 
 static unsigned int random_value = 1;
 static unsigned int random_stream = 0;
+static PyMutex random_mutex = {0};
```

Comment:
I was going to point out how using a shared lock means there would be a race on creating/initializing the lock.  However, that was before I saw this line and realized you are using `Py_Mutex`.  Good call!

FWIW, generally we should avoid using C global variables.  However, a lock that guards process-global resources is one of the few exceptions.  So +1 to this approach.